src/vs/workbench/services/extensionManagement/browser/webExtensionsScannerService.ts

```diff
@@ -696,13 +743,22 @@ export class WebExtensionsScannerService extends Disposable implements IWebExten
 			// Update
 			if (updateFn) {
```

Comment:
Can you please confirm if the migration is updating the resources in the file? Otherwise this migration runs everytime we read which will invalidate the purpose of caching the resources.

I think the migration has to happen outside `withWebExtensions` function. You shall have a migrate method that reads the web extensions and update them with migrated content.

Reply:
Ok I have moved this out of `withWebExtensions` one level higher.